app/liquid_tags/unified_embed.rb

```diff
@@ -0,0 +1,52 @@
+# The purpose of this singleton class is to provide a registry for the
+# numerous "link/embedded" type liquid tags.
+#
+# Each of those liquid tags must self-register their lookup regular
+# expression.
+class UnifiedEmbed
+  include Singleton
+
+  def self.current
+    instance
+  end
+
+  # @api public
+  #
+  # @param klass [Class] the LiquidTag class that we use when we match
+  #        the given :regexp
+  # @param regexp [Regexp] the regular expression that when matched
+  #        means we use the associated :klass
+  def self.register(klass, regexp:)
+    instance.register(klass, regexp: regexp)
+  end
+
+  # @api public
+  #
+  # @param link [String] the string that includes the URI for the
+  #        embed and possibly additional attributes, depending on how
+  #        the registered liquid tag parses this string.
+  # @return [Class] a descendant class of LiquidTagBase
+  def self.find_liquid_tag_for(link:)
+    instance.find_liquid_tag_for(link: link)
+  end
+
+  def initialize
+    @registry = []
```

Comment:
That was my inclination initially, but this structure allows us to register multiple REGEXP for a single class.  (We could do `@registry[klass] ||= []; @registry[klass] << regexp` but our logic becomes a bit more convoluted.

A "next" step is to consider how several of our custom liquid tags could instead "route" to a more generalized tag that uses open graph and twitter cards for embed.

Reply:
@jeremyf Do we have any concerns over linear-time lookup with an array vs constant-time with a hash?

Reply:
@jgaskins While we do have "a lot" of Liquid tags, their number is in the 10s, not 100s or 1000s, so I guess this wouldn't be the biggest issue.

@jeremyf 

>  this structure allows us to register multiple REGEXP for a single clas

I see. In this case a `regexp => klass` hash would have worked but given the limited number of tags (and considering we previously talked about removing some/turning them into plugins) instead of just adding more an array should definitely be ok here.

Reply:
@jgaskins we're treating the Hash as an Array, but the Hash gives some structure.

I'm uncertain how this will perform at large…but I'm hoping in sending things through a normalized process, we could start more generalization and allow Forem creators to more easily include their own.